core/controllers/domain_objects_validator.py

```diff
@@ -243,3 +243,67 @@ def validate_suggestion_images(files):
     # The files argument do not represent any domain class, hence dict form
     # of the data is returned from here.
     return files
+
+
+def validate_exploration_change(exp_change_dict):
+    """Validate the exploration change list
+
+    Args:
+        exp_change_dict: dict. The exploration change dictionary
+            that needs to be validated.
+
+    Returns:
+        exp_domain.ExplorationChange. The exploration change domain
+        object.
+    """
+    if exp_change_dict.get('cmd') is None:
```

Comment:
Can this be a classmethod in exp_domain.ExplorationChange class. Reasons:
- All the exp_changes related code in one place (Easy to maintain)

Reply:
Hey these changes are part of the controller layer validations, So I think these should be placed here only. Am I missing something here?

Reply:
If I'm not wrong this validation is used in the controller layer, right? If yes, then the validation checks can exist in the domain/service layer.

Reply:
Yes, they are for controller layer and will be called directly from the handler while validating the schema. I'm following the [wiki page](https://github.com/oppia/oppia/wiki/Schemas#domain-object-arguments-schemas-with-type-object_dict) for the validation and I think all the additional validations that we add for the Handlers are present inside this file only. I agree with your point that it can also be added to domain/service layer but the validations related to schemas or Handlers are added here in this file.
Thanks